application_sandbox/separate_depth_stencil_layouts/main.cpp

```diff
@@ -0,0 +1,716 @@
+        *frame_data->depth_stencil_image_view_,           // imageView
+        VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL  // imageLayout
+    };
+    cube_write.pBufferInfo = nullptr;
+    cube_write.descriptorCount = 1;
+    cube_write.descriptorType = VK_DESCRIPTOR_TYPE_INPUT_ATTACHMENT;
+    cube_write.pImageInfo = &image_info;
+    cube_write.dstBinding = 2;
+    app()->device()->vkUpdateDescriptorSets(app()->device(), 1, &cube_write, 0,
+                                            nullptr);
+
+    ::VkImageView raw_views[3] = {
+        color_view(frame_data), *frame_data->depth_stencil_image_view_,
+        *frame_data->depth_stencil_resolve_image_view_};
+
+    // Create a framebuffer with depth and image attachments
+    VkFramebufferCreateInfo framebuffer_create_info{
+        VK_STRUCTURE_TYPE_FRAMEBUFFER_CREATE_INFO,  // sType
+        nullptr,                                    // pNext
+        0,                                          // flags
+        *torus_render_pass_,                        // renderPass
+        3,                                          // attachmentCount
+        raw_views,                                  // attachments
+        app()->swapchain().width(),                 // width
+        app()->swapchain().height(),                // height
+        1                                           // layers
+    };
+
+    ::VkFramebuffer raw_framebuffer;
+    app()->device()->vkCreateFramebuffer(
+        app()->device(), &framebuffer_create_info, nullptr, &raw_framebuffer);
+    frame_data->framebuffer_ = containers::make_unique<vulkan::VkFramebuffer>(
+        data_->allocator(),
+        vulkan::VkFramebuffer(raw_framebuffer, nullptr, &app()->device()));
+
+    // Populate the render command buffer
+    frame_data->command_buffer_ =
+        containers::make_unique<vulkan::VkCommandBuffer>(
+            data_->allocator(), app()->GetCommandBuffer());
+
+    (*frame_data->command_buffer_)
+        ->vkBeginCommandBuffer((*frame_data->command_buffer_),
+                               &sample_application::kBeginCommandBuffer);
+    vulkan::VkCommandBuffer& cmdBuffer = (*frame_data->command_buffer_);
+
+    VkClearValue clears[3];
+    clears[0].color = {1.0f, 1.0f, 1.0f, 1.0f};
+    clears[1].depthStencil = {1.0f, 0};
+    clears[2].depthStencil = {1.0f, 0};
+
+    VkRenderPassBeginInfo pass_begin = {
+        VK_STRUCTURE_TYPE_RENDER_PASS_BEGIN_INFO,  // sType
+        nullptr,                                   // pNext
+        *torus_render_pass_,                       // renderPass
+        *frame_data->framebuffer_,                 // framebuffer
+        {{0, 0},
+         {app()->swapchain().width(),
+          app()->swapchain().height()}},  // renderArea
+        3,                                // clearValueCount
+        clears                            // clears
+    };
+
+    // Barrier before write to depth image
+    VkImageMemoryBarrier barrier{
```

Comment:
We do this everywhere, but I think the "better" way to solve this is have the renderpass start with an undefined image, and transition it to DEPTH_STENCIL_ATTACHMENT_OPTIMAL at first subpass, saves an explicit barrier.

That being said, we do this in most samples already.

Reply:
I think I will probably leave as is given samples already do this and barriers being used in this way are a part of the extension's features.